# Basic ticker history request -- returns normal price history

* let's stick with the time period parameter provided by yfinance instead of specifying start and end dates, this way there are just fewer potential things that can break. But if anybody has strong feelings, we could specify start and end dates.

In [10]:
# JSON request format:

# request = '{"ticker": "string", "period" (optional): "string", "interval" (optional): "string"}'

# use "period" instead of start/end
        # valid periods: 1d,5d,1mo,3mo,6mo,1y,2y,5y,10y,ytd,max
        # (optional, default is '1mo')

# fetch data by interval (including intraday if period < 60 days--
        # meaning, if we want a period greater than 60 days, the smallest interval available is 1d)
        # valid intervals: 1m,2m,5m,15m,30m,60m,90m,1h,1d,5d,1wk,1mo,3mo
        # (optional, default is '1d')

# example: 

ticker_request = '{"ticker": "BTC-USD", "period": "1d", "interval": "1h"}'


In [11]:
ticker_request

'{"ticker": "BTC-USD", "period": "1d", "interval": "1h"}'

In [12]:
# requests will be handled as follows.

import yfinance as yf
import pandas as pd
import json

request_dict = json.loads(ticker_request)
request_dict

{'ticker': 'BTC-USD', 'period': '1d', 'interval': '1h'}

In [13]:
ticker = yf.Ticker(request_dict['ticker'])
ticker_price = ticker.history(period=request_dict['period'], interval=request_dict['interval'])

In [16]:
ticker_price = ticker_price.drop(columns=['Dividends','Stock Splits'])
ticker_price

Open          High           Low  \
2022-03-08 00:00:00+00:00  38059.902344  38325.636719  37957.386719   
2022-03-08 01:00:00+00:00  38253.636719  38445.320312  38149.222656   
2022-03-08 02:00:00+00:00  38372.992188  38838.320312  38223.167969   
2022-03-08 03:00:00+00:00  38667.320312  38688.750000  38573.089844   
2022-03-08 04:00:00+00:00  38669.277344  38772.648438  38535.058594   
2022-03-08 04:54:00+00:00  38531.292969  38531.292969  38531.292969   

                                  Close     Volume  
2022-03-08 00:00:00+00:00  38266.105469  401766400  
2022-03-08 01:00:00+00:00  38402.375000  600453120  
2022-03-08 02:00:00+00:00  38674.261719  712640512  
2022-03-08 03:00:00+00:00  38665.253906   66793472  
2022-03-08 04:00:00+00:00  38535.058594  188155904  
2022-03-08 04:54:00+00:00  38531.292969          0

In [17]:
response = ticker_price.to_json(orient="table", date_format="iso")
response

'{"schema":{"fields":[{"name":"index","type":"datetime","tz":"UTC"},{"name":"Open","type":"number"},{"name":"High","type":"number"},{"name":"Low","type":"number"},{"name":"Close","type":"number"},{"name":"Volume","type":"integer"}],"primaryKey":["index"],"pandas_version":"0.20.0"},"data":[{"index":"2022-03-08T00:00:00.000Z","Open":38059.90234375,"High":38325.63671875,"Low":37957.38671875,"Close":38266.10546875,"Volume":401766400},{"index":"2022-03-08T01:00:00.000Z","Open":38253.63671875,"High":38445.3203125,"Low":38149.22265625,"Close":38402.375,"Volume":600453120},{"index":"2022-03-08T02:00:00.000Z","Open":38372.9921875,"High":38838.3203125,"Low":38223.16796875,"Close":38674.26171875,"Volume":712640512},{"index":"2022-03-08T03:00:00.000Z","Open":38667.3203125,"High":38688.75,"Low":38573.08984375,"Close":38665.25390625,"Volume":66793472},{"index":"2022-03-08T04:00:00.000Z","Open":38669.27734375,"High":38772.6484375,"Low":38535.05859375,"Close":38535.05859375,"Volume":188155904},{"index

### JSON reponse in following format, one entry as an example

sample_reponse = '{"schema":{"fields":[{"name":"index","type":"datetime","tz":"UTC"},
            {"name":"Open","type":"number"},
            {"name":"High","type":"number"},
            {"name":"Low","type":"number"},
            {"name":"Close","type":"number"},
            {"name":"Volume","type":"integer"}],
            "primaryKey":["index"],
                "pandas_version":"0.20.0"},
            "data":[{"index":"2022-03-08T00:00:00.000Z",
                     "Open":38059.90234375,
                     "High":38325.63671875,
                     "Low":37957.38671875,
                     "Close":38266.10546875,
                     "Volume":401766400}]}'


# ML prediction request -- returns normal price history as well as a prediction value for each time interval

* in progress